pkg/ccl/backupccl/show.go

```diff
@@ -93,54 +92,22 @@ func (m manifestInfoReader) header() colinfo.ResultColumns {
 // showBackup reads backup info from the manifest, populates the manifestInfoReader,
 // calls the backupShower to process the manifest info into datums,
 // and pipes the information to the user's sql console via the results channel.
-
-// TODO(msbutler): during the old backup syntax purge, remove store, incStore, incPaths,
-// and pass only `stores []cloud.ExternalStorage` object in signature
 func (m manifestInfoReader) showBackup(
 	ctx context.Context,
 	mem *mon.BoundAccount,
-	store cloud.ExternalStorage,
-	incStore cloud.ExternalStorage,
-	enc *jobspb.BackupEncryptionOptions,
-	incPaths []string,
-	manifestDirs []string,
+	mkStore cloud.ExternalStorageFromURIFactory,
+	info backupInfo,
+	user security.SQLUsername,
 	resultsCh chan<- tree.Datums,
 ) error {
-	var memSize int64
+	var (
+		err         error
+		memReserved int64
+	)
+
 	defer func() {
-		mem.Shrink(ctx, memSize)
+		mem.Shrink(ctx, memReserved)
```

Comment:
Ditto, does this work? Doesn't the closure make a copy?

Reply:
you're right!

Reply:
Are you sure this is in the right place now?